lib/measure_repository_service_test_kit/read_by_id.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require 'json'
+
+module MeasureRepositoryServiceTestKit
+  # tests for read by ID for Library and Measure services
+  class ReadById < Inferno::TestGroup
+    title 'Read by id'
+    description 'Ensure measure repository service can retrieve Measure and Library resources by the server-defined id'
+    id 'read_by_id'
+
+    fhir_client do
+      url :url
+    end
+
+    resource_type_options = { list_options: [{ label: 'Measure', value: 'Measure' },
+                                             { label: 'Library', value: 'Library' }] }
+    resource_type_args = { type: 'radio', optional: false, default: 'Measure', options: resource_type_options,
+                           title: 'Resource Type' }
+
+    INVALID_ID = 'INVALID_ID'
+
+    test do
+      title 'Server returns 200 response status and correct Measure/Library resource from the read interaction'
+      id 'read-by-id-01'
+      description %(This test verifies that the Measure and Library resources can be read from the server.)
+      input :resource_type, **resource_type_args
+      input :resource_id, title: 'Resource id'
+
+      run do
+        fhir_read(resource_type, resource_id)
+
+        assert_response_status(200)
+        assert_resource_type(resource_type)
+        assert resource.id == resource_id,
```

Comment:
not sure how crucial this is, but in DEQM test kit, we always did an `assert_valid_json(response[:body])` here. Could be worth adding for more explicit/helpful error reporting if that case gets hit.

Reply:
Good point, I think we should add it!